### ISSUE TITLE
Alpha12 with mingw "cp: cannot create regular file 'test/': Not a directory"

### ISSUE THREAD

**Issue by misery** [openssl/openssl#14246]
Now we tried alpha12 in our CI farm. MinGW has a problem now. Seems there is a mistake with a "test" directory.

**Using: i686-7.3.0-release-posix-dwarf-rt_v5-rev0**

### CI output
```
11:16:25 ../openssl/crypto/init.c: In function 'OPENSSL_atexit':
11:16:25 ../openssl/crypto/init.c:668:11: warning: variable 'handlersym' set but not used [-Wunused-but-set-variable]
11:16:25          } handlersym;
11:16:25            ^~~~~~~~~~

11:16:29 ../openssl/providers/implementations/encode_decode/encode_key2text.c: In function 'rsa_to_text':
11:16:29 ../openssl/providers/implementations/encode_decode/encode_key2text.c:686:10: warning: 'exponent_label' may be used uninitialized in this function [-Wmaybe-uninitialized]
11:16:29      if (!print_labeled_bignum(out, exponent_label, rsa_e))
11:16:29           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
11:16:29 ../openssl/providers/implementations/encode_decode/encode_key2text.c:684:10: warning: 'modulus_label' may be used uninitialized in this function [-Wmaybe-uninitialized]
11:16:29      if (!print_labeled_bignum(out, modulus_label, rsa_n))
11:16:29           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
11:16:31 ../openssl/ssl/t1_enc.c: In function 'tls1_setup_key_block':
11:16:31 ../openssl/ssl/t1_enc.c:567:50: warning: format '%ld' expects argument of type 'long int', but argument 3 has type 'size_t {aka unsigned int}' [-Wformat=]
11:16:31          BIO_printf(trc_out, "key block length: %ld\n", num);
11:16:31                                                 ~~^
11:16:31                                                 %d

11:16:37 ../openssl/crypto/init.c: In function 'OPENSSL_atexit':
11:16:37 ../openssl/crypto/init.c:668:11: warning: variable 'handlersym' set but not used [-Wunused-but-set-variable]
11:16:37          } handlersym;
11:16:37            ^~~~~~~~~~

11:16:41 ../openssl/ssl/t1_enc.c: In function 'tls1_setup_key_block':
11:16:41 ../openssl/ssl/t1_enc.c:567:50: warning: format '%ld' expects argument of type 'long int', but argument 3 has type 'size_t {aka unsigned int}' [-Wformat=]
11:16:41          BIO_printf(trc_out, "key block length: %ld\n", num);
11:16:41                                                 ~~^
11:16:41                                                 %d

11:16:54 cp: cannot create regular file 'test/': Not a directory
11:16:54 mingw32-make[5]: *** [Makefile:2286: libcrypto-3.dll] Error 1
11:16:54 mingw32-make[4]: *** [Makefile:1295: build_sw] Error 2
11:16:54 mingw32-make[3]: *** [CMakeFiles\openssl.dir\build.make:134: b/src/openssl-stamp/openssl-build] Error 2
11:16:54 mingw32-make[2]: *** [CMakeFiles\Makefile2:98: CMakeFiles/openssl.dir/all] Error 2
11:16:54 mingw32-make[1]: *** [CMakeFiles\Makefile2:105: CMakeFiles/openssl.dir/rule] Error 2
```

### Config
```
    perl ../openssl/Configure --prefix=R:/J/workspace/default_Review_Libs_Win32_GNU/build/dist no-ui-console no-filenames no-camellia no-bf no-aria no-seed no-poly1305 no-srp no-gost no-idea no-mdc2 no-rc2 no-rc4 no-rc5 no-srtp no-sm2 no-sm3 no-sm4 no-ct no-dgram no-cast no-chacha no-blake2 no-rmd160 no-scrypt no-siphash no-whirlpool no-md4 no-des no-tls1 no-tls1-method no-tls1_1 no-tls1_1-method no-tls1_3 no-ssl3 no-ssl3-method no-dtls no-dtls1-method no-dtls1_2-method no-engine no-async no-dso no-comp no-ts no-makedepend no-tests no-legacy shared mingw -Os -fstack-protector-strong

Perl information:

    perl
    5.28.1 for x86_64-msys-thread-multi

Enabled features:

    acvp_tests
    asm
    autoalginit
    autoerrinit
    autoload-config
    bulk
    cached-fetch
    cmac
    cmp
    cms
    deprecated
    dh
    dsa
    ec
    ec2m
    ecdh
    ecdsa
    err
    fips-securitychecks
    multiblock
    nextprotoneg
    pinshared
    ocb
    ocsp
    pic
    posix-io
    psk
    rdrand
    rfc3779
    secure-memory
    shared
    siv
    sock
    sse2
    ssl
    static-engine
    stdio
    threads
    tls
    uplink
    tls1_2
    tls1_2-method

Disabled features:

    afalgeng            [cascade]                OPENSSL_NO_AFALGENG
    aria                [option]                 OPENSSL_NO_ARIA (skip crypto/aria)
    asan                [default]                OPENSSL_NO_ASAN
    async               [option]                 OPENSSL_NO_ASYNC
    bf                  [option]                 OPENSSL_NO_BF (skip crypto/bf)
    blake2              [option]                 OPENSSL_NO_BLAKE2
    buildtest-c++       [default]                
    camellia            [option]                 OPENSSL_NO_CAMELLIA (skip crypto/camellia)
    capieng             [cascade]                OPENSSL_NO_CAPIENG
    cast                [option]                 OPENSSL_NO_CAST (skip crypto/cast)
    chacha              [option]                 OPENSSL_NO_CHACHA (skip crypto/chacha)
    comp                [option]                 OPENSSL_NO_COMP (skip crypto/comp)
    crypto-mdebug       [default]                OPENSSL_NO_CRYPTO_MDEBUG
    ct                  [option]                 OPENSSL_NO_CT (skip crypto/ct)
    des                 [option]                 OPENSSL_NO_DES (skip crypto/des)
    devcryptoeng        [default]                OPENSSL_NO_DEVCRYPTOENG
    dgram               [option]                 OPENSSL_NO_DGRAM
    dso                 [option]                 OPENSSL_NO_DSO
    dtls                [option(dtls)]           OPENSSL_NO_DTLS
    dynamic-engine      [cascade]                
    ec_nistp_64_gcc_128 [default]                OPENSSL_NO_EC_NISTP_64_GCC_128
    egd                 [default]                OPENSSL_NO_EGD
    engine              [option]                 OPENSSL_NO_ENGINE (skip engines, crypto/engine)
    external-tests      [default]                OPENSSL_NO_EXTERNAL_TESTS
    filenames           [option]                 OPENSSL_NO_FILENAMES
    fips                [cascade]                
    fuzz-libfuzzer      [default]                OPENSSL_NO_FUZZ_LIBFUZZER
    fuzz-afl            [default]                OPENSSL_NO_FUZZ_AFL
    gost                [option]                 OPENSSL_NO_GOST
    idea                [option]                 OPENSSL_NO_IDEA (skip crypto/idea)
    ktls                [default]                OPENSSL_NO_KTLS
    legacy              [option]                 
    makedepend          [option]                 
    md2                 [default]                OPENSSL_NO_MD2 (skip crypto/md2)
    md4                 [option]                 OPENSSL_NO_MD4 (skip crypto/md4)
    mdc2                [option]                 OPENSSL_NO_MDC2 (skip crypto/mdc2)
    module              [cascade]                
    msan                [default]                OPENSSL_NO_MSAN
    padlockeng          [cascade]                OPENSSL_NO_PADLOCKENG
    poly1305            [option]                 OPENSSL_NO_POLY1305 (skip crypto/poly1305)
    rc2                 [option]                 OPENSSL_NO_RC2 (skip crypto/rc2)
    rc4                 [option]                 OPENSSL_NO_RC4 (skip crypto/rc4)
    rc5                 [option]                 OPENSSL_NO_RC5 (skip crypto/rc5)
    rmd160              [option]                 OPENSSL_NO_RMD160 (skip crypto/ripemd)
    scrypt              [option]                 OPENSSL_NO_SCRYPT
    sctp                [default]                OPENSSL_NO_SCTP
    seed                [option]                 OPENSSL_NO_SEED (skip crypto/seed)
    siphash             [option]                 OPENSSL_NO_SIPHASH (skip crypto/siphash)
    sm2                 [option]                 OPENSSL_NO_SM2 (skip crypto/sm2)
    sm3                 [option]                 OPENSSL_NO_SM3 (skip crypto/sm3)
    sm4                 [option]                 OPENSSL_NO_SM4 (skip crypto/sm4)
    srp                 [option]                 OPENSSL_NO_SRP (skip crypto/srp)
    srtp                [option]                 OPENSSL_NO_SRTP
    ssl-trace           [default]                OPENSSL_NO_SSL_TRACE
    tests               [option]                 OPENSSL_NO_TESTS
    trace               [default]                OPENSSL_NO_TRACE
    ts                  [option]                 OPENSSL_NO_TS (skip crypto/ts)
    ubsan               [default]                OPENSSL_NO_UBSAN
    ui-console          [option]                 OPENSSL_NO_UI_CONSOLE
    unit-test           [default]                OPENSSL_NO_UNIT_TEST
    whirlpool           [option]                 OPENSSL_NO_WHIRLPOOL (skip crypto/whrlpool)
    weak-ssl-ciphers    [default]                OPENSSL_NO_WEAK_SSL_CIPHERS
    zlib                [default]                
    zlib-dynamic        [default]                
    ssl3                [option(ssl3-method)]    OPENSSL_NO_SSL3
    ssl3-method         [option]                 OPENSSL_NO_SSL3_METHOD
    tls1                [option(tls1-method)]    OPENSSL_NO_TLS1
    tls1-method         [option]                 OPENSSL_NO_TLS1_METHOD
    tls1_1              [option(tls1_1-method)]  OPENSSL_NO_TLS1_1
    tls1_1-method       [option]                 OPENSSL_NO_TLS1_1_METHOD
    tls1_3              [option]                 OPENSSL_NO_TLS1_3
    dtls1               [option(dtls1-method)]   OPENSSL_NO_DTLS1
    dtls1-method        [option]                 OPENSSL_NO_DTLS1_METHOD
    dtls1_2             [option(dtls1_2-method)] OPENSSL_NO_DTLS1_2
    dtls1_2-method      [option]                 OPENSSL_NO_DTLS1_2_METHOD

Config target attributes:

    AR => "ar",
    ARFLAGS => "qc",
    CC => "gcc",
    CFLAGS => "-Wall -O3 -fomit-frame-pointer",
    HASHBANGPERL => "/usr/bin/env perl",
    RANLIB => "ranlib",
    RC => "windres",
    asm_arch => "x86",
    bn_ops => "BN_LLONG",
    build_file => "Makefile",
    build_scheme => [ "unified", "unix" ],
    cflags => "-m32",
    cppflags => "-DUNICODE -D_UNICODE -DWIN32_LEAN_AND_MEAN -D_MT",
    defines => [ "OPENSSL_BUILDING_OPENSSL" ],
    disable => [  ],
    dso_scheme => "win32",
    enable => [  ],
    ex_libs => "-lws2_32 -lgdi32 -lcrypt32",
    includes => [  ],
    lflags => "",
    lib_cflags => "",
    lib_cppflags => "-DL_ENDIAN",
    lib_defines => [  ],
    module_cflags => "",
    module_cxxflags => undef,
    module_ldflags => "-static-libgcc -shared -Wl,--enable-auto-image-base",
    multilib => "",
    perl_platform => "mingw",
    perlasm_scheme => "coff",
    shared_argfileflag => "\@",
    shared_cflag => "",
    shared_cppflags => "_WINDLL",
    shared_defflag => "",
    shared_defines => [  ],
    shared_impflag => "-Wl,--out-implib=",
    shared_ldflag => "-static-libgcc -shared -Wl,--enable-auto-image-base",
    shared_rcflag => "--target=pe-i386",
    shared_target => "mingw-shared",
    sys_id => "MINGW32",
    thread_defines => [  ],
    thread_scheme => "winthreads",
    unistd => "<unistd.h>",
    uplink_arch => "x86",

Recorded environment:

    AR = 
    BUILDFILE = 
    CC = 
    CFLAGS = 
    CPPFLAGS = 
    CROSS_COMPILE = 
    CXX = 
    CXXFLAGS = 
    HASHBANGPERL = 
    LDFLAGS = 
    LDLIBS = 
    OPENSSL_LOCAL_CONFIG_DIR = 
    PERL = 
    RANLIB = 
    RC = 
    RCFLAGS = 
    WINDRES = 
    __CNF_CFLAGS = 
    __CNF_CPPDEFINES = 
    __CNF_CPPFLAGS = 
    __CNF_CPPINCLUDES = 
    __CNF_CXXFLAGS = 
    __CNF_LDFLAGS = 
    __CNF_LDLIBS = 

Makevars:

    AR              = ar
    ARFLAGS         = qc
    CC              = gcc
    CFLAGS          = -Wall -O3 -fomit-frame-pointer -Os -fstack-protector-strong
    CPPDEFINES      = 
    CPPFLAGS        = 
    CPPINCLUDES     = 
    CXXFLAGS        = -Os -fstack-protector-strong
    HASHBANGPERL    = /usr/bin/env perl
    LDFLAGS         = 
    LDLIBS          = 
    PERL            = perl
    RANLIB          = ranlib
    RC              = windres
    RCFLAGS         = 
```

**Comment by mattcaswell** [comment]
This seems to be the combination of a mingw target, an out-of-tree build and the "no-tests" configure option that causes this.

I was able to replicate this on Linux as follows:

````
mkdir _build
cd _build
perl ../Configure --cross-compile-prefix=i686-w64-mingw32- no-tests mingw
make -j
````

Perhaps @levitte has an idea why?

**Comment by mattcaswell** [comment]
This failure with "no-tests" seems to have been introduced by commit e60a748a13 (PR #14082). Ironically the commit has the title "ensure that 'no-tests' works correctly".

**Comment by mattcaswell** [comment]
Fix in #14249
